core/src/main/java/smile/manifold/UMAP.java

```diff
@@ -0,0 +1,559 @@
+/*******************************************************************************
+ * Copyright (c) 2010-2020 Haifeng Li. All rights reserved.
+ *
+ * Smile is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * Smile is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Smile.  If not, see <https://www.gnu.org/licenses/>.
+ ******************************************************************************/
+
+package smile.manifold;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import smile.graph.Graph;
+import smile.graph.Graph.Edge;
+import smile.math.DifferentiableMultivariateFunction;
+import smile.math.LevenbergMarquardt;
+import smile.math.MathEx;
+import smile.math.distance.Distance;
+import smile.math.distance.EuclideanDistance;
+import smile.math.matrix.DenseMatrix;
+import smile.math.matrix.EVD;
+import smile.math.matrix.SparseMatrix;
+import smile.netlib.ARPACK;
+
+/**
+ * 
```

Comment:
Add first sentence "Uniform Manifold Approximation and Projection."

It is a good practice to have a short sentence at the beginning, which will be shown in the package page of generated documents.